etna/transforms/change_points_trend.py

```diff
@@ -0,0 +1,204 @@
+        timestamp = series.index
+        self.change_point_model.fit(signal=signal)
+        # last point in change points is the first index after the series
+        change_points_indices = self.change_point_model.predict(**self.change_point_model_predict_params)[:-1]
+        change_points = [timestamp[idx] for idx in change_points_indices]
+        return change_points
+
+    @staticmethod
+    def _build_trend_intervals(change_points: List[pd.Timestamp]) -> List[TTimestampInterval]:
+        """Create list of stable trend intervals from list of change points."""
+        change_points = sorted(change_points)
+        left_border = pd.Timestamp.min
+        intervals = []
+        for point in change_points:
+            right_border = point
+            intervals.append((left_border, right_border))
+            left_border = right_border
+        intervals.append((left_border, pd.Timestamp.max))
+        return intervals
+
+    def _init_detrend_models(
+        self, intervals: List[TTimestampInterval]
+    ) -> Dict[Tuple[pd.Timestamp, pd.Timestamp], TDetrendModel]:
+        """Create copy of detrend model for each timestamp interval."""
+        per_interval_models = {interval: deepcopy(self.detrend_model) for interval in intervals}
+        return per_interval_models
+
+    def _get_timestamps(self, series: pd.Series) -> np.ndarray:
+        """Convert ETNA timestamp-index to a list of timestamps to fit regression models."""
+        timestamps = series.index
+        timestamps = np.array([[ts.timestamp()] for ts in timestamps])
+        return timestamps
+
+    def _fit_per_interval_model(self, series: pd.Series):
+        """Fit per-interval models with corresponding data from series."""
+        for interval in self.intervals:
+            tmp_series = series[interval[0] : interval[1]]
+            x = self._get_timestamps(series=tmp_series)
+            y = tmp_series.values
+            self.per_interval_models[interval].fit(x, y)
+
+    def _predict_per_interval_model(self, series: pd.Series) -> pd.Series:
+        """Apply per-interval detrending to series."""
+        trend_series = pd.Series(index=series.index)
+        for interval in self.intervals:
+            tmp_series = series[interval[0] : interval[1]]
+            if tmp_series.empty:
+                continue
+            x = self._get_timestamps(series=tmp_series)
+            trend = self.per_interval_models[interval].predict(x)
+            trend_series[tmp_series.index] = trend
+        return trend_series
+
+    def fit(self, df: pd.DataFrame) -> "OneSegmentChangePointsTransform":
+        """Fit OneSegmentChangePointsTransform: find thrend change points in df, fit detrend models with data from intervals of stable trend.
+
+        Parameters
+        ----------
+        df:
+            one segment dataframe indexed with timestamp
+
+        Returns
+        -------
+        self
+        """
+        series = df[self.in_column].dropna()
```

Comment:
Maybe it's too strong assumtion? And we should remove nan's only at the beginning? 
 This `self.change_point_model.fit(signal=signal)` may not work correctly

Reply:
fixed